src/Libraries/RevitNodes/Elements/ElementType.cs

```diff
@@ -0,0 +1,200 @@
+        {
+            get { return InternalElementType.CanBeCopied; }
+        }
+
+        /// <summary>
+        /// Determine if this ElementType can be renamed
+        /// </summary>
+        public bool CanBeRenamed
+        {
+            get { return InternalElementType.CanBeRenamed; }
+        }
+
+        #endregion
+
+        #region Public constructors
+
+        /// <summary>
+        /// Duplicates an existing element type and assigns it a new name.
+        /// </summary>
+        /// <param name="name">The new name of the element type.</param>
+        /// <returns>The duplicated element type.</returns>
+        public ElementType Duplicate(string name)
+        {
+            if (String.IsNullOrWhiteSpace(name))
+                throw new ArgumentNullException(nameof(name));
+
+            TransactionManager.Instance.EnsureInTransaction(Document);
+            ElementType newElementType = FromExisting(this.InternalElementType.Duplicate(name), true);
+            TransactionManager.Instance.TransactionTaskDone();
+            return newElementType;  
+        }
+
+        #endregion
+
+        #region Public static constructors
+
+        /// <summary>
+        /// Returns the type Element with the given name.
+        /// </summary>
+        /// <param name="name">Name of the type</param>
+        /// <returns>Type Element</returns>
+        public static ElementType ByName(string name)
+        {
+            if (String.IsNullOrWhiteSpace(name))
+                throw new ArgumentNullException(nameof(name));
+
+            var elementType = DocumentManager.Instance
+                .ElementsOfType<Autodesk.Revit.DB.ElementType>()
+                .FirstOrDefault(x => x.Name == name);
+
+            if (elementType == null)
+                throw new KeyNotFoundException(Properties.Resources.ElementTypeNameNotFound);
+            return FromExisting(elementType, true);
+        }
+
+        #endregion
+
+        #region Internal static constructors
+
+        /// <summary>
+        /// Create a ElementType from a user selected Element.
+        /// </summary>
+        /// <param name="elementType"></param>
+        /// <param name="isRevitOwned"></param>
+        /// <returns></returns>
+        internal static ElementType FromExisting(Autodesk.Revit.DB.ElementType elementType, bool isRevitOwned)
+        {
+            return new ElementType(elementType)
+            {
+                IsRevitOwned = isRevitOwned
+            };
+        }
+
+        #endregion
+
+        #region Public methods
+
+        /// <summary>
+        /// Get the preview image of an element. This image is similar to what is seen in
+        /// the Revit UI when selecting the type of an element.
+        /// </summary>
+        /// <param name="size">The width and height of the preview image in pixels.</param>
+        /// <returns>The preview image. null if there is no preview image.</returns>
+        public Bitmap GetPreviewImage(int size)
```

Comment:
should we have a default size here ? As a user, I wouldn't know what number to input here at first, i'd maybe try something like `256` (a power of 2 number).

Let's check if the `GetPreviewImage` defaults to something ?